### PR TITLE
Improve assembly resolution in ResultChecker

### PR DESCRIPTION
**Pull request by mrvoorhe**
Add support for string type name in *InAssemblyAttributes so that non-public types can be checked

### CODE REVIEW

File: linker/Tests/TestCasesRunner/ResultChecker.cs [hunk 1]
```diff
@@ -7,31 +7,68 @@
...
-			readerParams.AssemblyResolver = resolver;
-			resolver.AddSearchDirectory (assemblyPath.Parent.ToString ());
-			return new AssemblyContainer (AssemblyDefinition.ReadAssembly (assemblyPath.ToString (), readerParams), resolver);
+			return _originalsResolver.Resolve (new AssemblyNameReference (assemblyPath.FileNameWithoutExtension, null));
+		}
+
+		AssemblyDefinition ResolveOriginalsAssembly (string assemblyName)
+		{
+			var cleanAssemblyName = assemblyName;
+			if (assemblyName.EndsWith (".exe") || assemblyName.EndsWith (".dll"))
```
Comment (marek-safar) [inline linker/Tests/TestCasesRunner/ResultChecker.cs]: Use Path.GetFileNameWithoutExtension

File: linker/Tests/Mono.Linker.Tests.Cases.Expectations/Assertions/KeptMemberInAssemblyAttribute.cs [hunk 2]
```diff
@@ -5,13 +5,20 @@ namespace Mono.Linker.Tests.Cases.Expectations.Assertions {
 	[AttributeUsage (AttributeTargets.Class | AttributeTargets.Delegate, AllowMultiple = true, Inherited = false)]
 	public class KeptMemberInAssemblyAttribute : BaseExpectedLinkedBehaviorAttribute {
 		public readonly string AssemblyFileName;
```
Comment (marek-safar) [inline linker/Tests/Mono.Linker.Tests.Cases.Expectations/Assertions/KeptMemberInAssemblyAttribute.cs]: Maybe we could just drop the fields completely, we are not using them anyway

Comment (mrvoorhe) [inline linker/Tests/Mono.Linker.Tests.Cases.Expectations/Assertions/KeptMemberInAssemblyAttribute.cs]: I've been on the fence about what to do as well.  My concern with dropping the fields is that then the constructor arguments will be highlighted in the IDE as not used and that would be annoying in it's own way.

File: linker/Tests/TestCasesRunner/ResultChecker.cs (hunk 1 above)
Comment (mrvoorhe) [inline linker/Tests/TestCasesRunner/ResultChecker.cs]: Done

File: linker/Tests/Mono.Linker.Tests.Cases.Expectations/Assertions/KeptMemberInAssemblyAttribute.cs (hunk 2 above)
Comment (marek-safar) [inline linker/Tests/Mono.Linker.Tests.Cases.Expectations/Assertions/KeptMemberInAssemblyAttribute.cs]: I am not really concern about IDE but we do something like Debug.Assert (...) or null check with throw

Comment (mrvoorhe) [inline linker/Tests/Mono.Linker.Tests.Cases.Expectations/Assertions/KeptMemberInAssemblyAttribute.cs]: Ok.  I will make the change to all of the attributes in a separate PR

**Pull request merged**